src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernelExtension.cs

```diff
@@ -1,183 +1,190 @@
-                ExecutionContext.SuppressFlow();
-                try
-                {
-                    FormatHttpResponseMessageAsHtml(
-                        responseMessage,
-                        context).Wait();
-                }
-                finally
-                {
-                    ExecutionContext.RestoreFlow();
-                }
-
-                return true;
-            }, HtmlFormatter.MimeType);
-        }
-        return Task.CompletedTask;
-    }
-
-    private static async Task FormatHttpResponseMessageAsPlainText(
-        HttpResponseMessage responseMessage,
-        FormatContext context)
-    {
-        var responseBodyString = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
-
-        context.Writer.WriteLine($"Status Code: {(int)responseMessage.StatusCode} {responseMessage.StatusCode}");
-        context.Writer.WriteLine($"Request URI: {responseMessage.RequestMessage.RequestUri}");
-        context.Writer.WriteLine($"Content type: {responseMessage.Content.Headers.ContentType}");
-        foreach (var header in responseMessage.Headers)
-        {
-            context.Writer.WriteLine($"{header.Key}: {string.Join(", ", header.Value)}");
-        }
-
-        if (!string.IsNullOrWhiteSpace(responseBodyString))
-        {
-            switch (responseMessage.Content.Headers.ContentType.MediaType)
-            {
-                case "application/json":
-                case "text/json":
-                    var formatted = JsonSerializer.Serialize(JsonDocument.Parse(responseBodyString).RootElement, new JsonSerializerOptions { WriteIndented = true });
-                    context.Writer.WriteLine($"Body: {formatted}");
-                    break;
-                default:
-                    context.Writer.WriteLine($"Body: {responseBodyString}");
-                    break;
-            }
-        }
-    }
-
-
-    private const string _containerClass = "http-response-message-container";
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.DotNet.Interactive.Formatting;
+
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest;
+
+public class HttpRequestKernelExtension 
+{
+    public static void  Load(Kernel kernel)
+    {
+        if (kernel.RootKernel is CompositeKernel compositeKernel)
+        {
+            var httpRequestKernel = new HttpRequestKernel();
+            compositeKernel.Add(httpRequestKernel);
+            httpRequestKernel.UseValueSharing();
+            
+            Formatter.Register<HttpResponseMessage>((responseMessage, context) =>

```

Comment:
Are these not redundant with the ones already provided?

Reply:
removed